client/src/actions/createWalletActions.js

```diff
@@ -17,6 +17,13 @@ const generateKeyphrase = () => {
 	};
 };
 
+const updateKeyphrase = (keyphrase) => {
```

Comment:
We already have `generateKeyphrase` which dispatches `CREATE_WALLET.CREATE_KEYPHRASE` when creating new wallet, so in order to keep this code dry. My suggestion is as following:

// Convert normal action creator to thunk
```
const updateKeyphrase = keyphrase => {
	return {
		type: CREATE_WALLET.CREATE_KEYPHRASE,
		payload: { keyphrase, map: generateKeyphraseMap(keyphrase) },
	};
};

const generateKeyphrase = () => {
	const keyManager = KeyFactory.getInstance();
	const keyphrase = keyManager.generate();

	return dispatch => dispatch(updateKeyphrase(keyphrase));
};
```